File: ios-native/ClubFuoco/Core/DesignSystem/Components.swift
import SwiftUI

// Shared building blocks for the auth surfaces — native counterparts of
// AuthField / PrimaryBtn / ProgressBar in the (auth) pages.

/// Labeled white rounded field container (mono uppercase label above).
struct AuthField<Content: View>: View {
    let label: String
    var error: Bool = false
    @ViewBuilder var content: Content

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Kicker(label, color: Theme.fadedSand, size: 9)
            HStack(spacing: 8) {
                content
            }
            .padding(.horizontal, 16)
            .frame(minHeight: 50)
            .background(
                error ? Theme.wine.opacity(0.04) : Color.white,
                in: .rect(cornerRadius: 12)
            )
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .stroke(error ? Theme.wine : Theme.ink.opacity(0.08))
            )
        }
    }
}

/// Wine-red primary CTA with trailing arrow, "Please wait…" while loading.
struct PrimaryButton: View {
    let title: String
    var loading: Bool = false
    var disabled: Bool = false
    var background: Color = Theme.wine
    let action: () -> Void

    @Environment(LocaleStore.self) private var locale

    var body: some View {
        Button {
            Haptics.tap()
            action()
        } label: {
            HStack(spacing: 8) {
                if loading {
                    Text(locale.t("auth.pleaseWait"))
                        .font(.cfSans(15, weight: .medium))
                } else {
                    Text(title)
                        .font(.cfSans(15, weight: .medium))
                    Image(systemName: "arrow.right")
                        .font(.system(size: 13, weight: .medium))
                }
            }
            .frame(maxWidth: .infinity)
            .frame(height: 55)
            .background(
                background.opacity(disabled && !loading ? 0.3 : loading ? 0.7 : 1),
                in: .rect(cornerRadius: Theme.radiusField)
            )
            .foregroundStyle(Theme.cream)
        }
        .disabled(disabled || loading)
    }
}

/// Thin segmented progress bar (signup wizard).
struct SegmentedProgress: View {
    let step: Int
    let total: Int

    var body: some View {
        HStack(spacing: 6) {
            ForEach(0..<total, id: \.self) { i in
                Capsule()
                    .fill(i < step ? Theme.wine : Theme.ink.opacity(0.16))
                    .frame(height: 2)
            }
        }
        .animation(.easeInOut(duration: 0.3), value: step)
    }
}

/// Mono uppercase "← BACK" header button.
struct BackChevronButton: View {
    let action: () -> Void
    @Environment(LocaleStore.self) private var locale

    var body: some View {
        Button(action: action) {
            HStack(spacing: 6) {
                Image(systemName: "chevron.left")
                    .font(.system(size: 11, weight: .medium))
                Text(locale.t("auth.back").uppercased())
                    .font(.cfMono(10))
                    .kerning(1.8)
            }
            .foregroundStyle(Theme.stone)
        }
    }
}

/// Inline error message in brand red.
struct FormError: View {
    let message: String

    var body: some View {
        Text(message)
            .font(.cfSans(12))
            .foregroundStyle(Theme.wine)
            .frame(maxWidth: .infinity, alignment: .leading)
    }
}

File: android-native/app/src/main/java/com/clubfuoco/app/core/designsystem/Components.kt
package com.clubfuoco.app.core.designsystem

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.defaultMinSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowForward
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowLeft
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.clubfuoco.app.R

/**
 * Shared building blocks for the auth surfaces — port of `Components.swift`,
 * itself the native counterpart of AuthField / PrimaryBtn / ProgressBar in the
 * web (auth) pages.
 */

/** Small uppercase monospaced overline used across the brand. */
@Composable
fun Kicker(
    text: String,
    modifier: Modifier = Modifier,
    color: Color = Theme.wine,
    size: TextUnit = 9.5.sp,
) {
    Text(
        text = text.uppercase(),
        modifier = modifier,
        fontFamily = GeistMono,
        fontSize = size,
        letterSpacing = 2.0.sp,
        color = color,
    )
}

/** Labelled rounded field container with a mono uppercase label above. */
@Composable
fun AuthField(
    label: String,
    modifier: Modifier = Modifier,
    error: Boolean = false,
    content: @Composable () -> Unit,
) {
    Column(modifier, verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Kicker(label, color = Theme.fadedSand, size = 9.sp)
        Row(
            Modifier
                .fillMaxWidth()
                .defaultMinSize(minHeight = 50.dp)
                .clip(RoundedCornerShape(12.dp))
                .background(if (error) Theme.wine.copy(alpha = 0.04f) else Theme.surface)
                .border(
                    BorderStroke(1.dp, if (error) Theme.wine else Theme.ink.copy(alpha = 0.08f)),
                    RoundedCornerShape(12.dp),
                )
                .padding(horizontal = 16.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            content()
        }
    }
}

/** Wine-red primary CTA with trailing arrow; "Please wait…" while loading. */
@Composable
fun PrimaryButton(
    title: String,
    modifier: Modifier = Modifier,
    loading: Boolean = false,
    enabled: Boolean = true,
    background: Color = Theme.wine,
    onClick: () -> Unit,
) {
    val disabled = !enabled
    val alpha = when {
        disabled && !loading -> 0.3f
        loading -> 0.7f
        else -> 1f
    }
    Box(
        modifier
            .fillMaxWidth()
            .height(55.dp)
            .clip(RoundedCornerShape(FuocoRadius.field))
            .background(background.copy(alpha = alpha))
            .clickableUnlessBusy(enabled = enabled && !loading, onClick = onClick),
        contentAlignment = Alignment.Center,
    ) {
        Row(
            horizontalArrangement = Arrangement.spacedBy(8.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            if (loading) {
                Text(
                    stringResource(R.string.auth_pleaseWait),
                    fontFamily = Geist,
                    fontWeight = FontWeight.Medium,
                    fontSize = 15.sp,
                    color = Theme.cream,
                )
            } else {
                Text(
                    title,
                    fontFamily = Geist,
                    fontWeight = FontWeight.Medium,
                    fontSize = 15.sp,
                    color = Theme.cream,
                )
                Icon(
                    Icons.AutoMirrored.Filled.ArrowForward,
                    contentDescription = null,
                    tint = Theme.cream,
                    modifier = Modifier.size(16.dp),
                )
            }
        }
    }
}

/** Thin segmented progress bar for the signup wizard. */
@Composable
fun SegmentedProgress(step: Int, total: Int, modifier: Modifier = Modifier) {
    Row(modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(6.dp)) {
        repeat(total) { index ->
            Box(
                Modifier
                    .weight(1f)
                    .height(2.dp)
                    .clip(CircleShape)
                    .background(if (index < step) Theme.wine else Theme.ink.copy(alpha = 0.16f)),
            )
        }
    }
}

/** Mono uppercase "← BACK" header button. */
@Composable
fun BackChevronButton(modifier: Modifier = Modifier, onClick: () -> Unit) {
    Row(
        modifier.clickableUnlessBusy(onClick = onClick),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(6.dp),
    ) {
        Icon(
            Icons.AutoMirrored.Filled.KeyboardArrowLeft,
            contentDescription = null,
            tint = Theme.stone,
            modifier = Modifier.size(16.dp),
        )
        Text(
            stringResource(R.string.auth_back).uppercase(),
            fontFamily = GeistMono,
            fontSize = 10.sp,
            letterSpacing = 1.8.sp,
            color = Theme.stone,
        )
    }
}

/** Inline error message in brand red. */
@Composable
fun FormError(message: String, modifier: Modifier = Modifier) {
    Text(
        message,
        modifier = modifier.fillMaxWidth(),
        fontFamily = Geist,
        fontSize = 12.sp,
        color = Theme.wine,
    )
}
